include/gui/gui_api/gui_api.h

```diff
@@ -0,0 +1,82 @@
+    void select_gate(std::shared_ptr<gate> gate, bool clear_current_selection = true);
+    void select_gate(std::vector<std::shared_ptr<gate>> gates, bool clear_current_selection = true);
+
+    void select_net(u32 net_id, bool clear_current_selection = true);
+    void select_net(std::vector<u32> net_ids, bool clear_current_selection = true);
+    void select_net(std::shared_ptr<net> net, bool clear_current_selection = true);
+    void select_net(std::vector<std::shared_ptr<net>> nets, bool clear_current_selection = true);
+
+    void select_module(u32 module_id, bool clear_current_selection = true);  
+    void select_module(std::vector<u32> module_ids, bool clear_current_selection = true);
+    void select_module(std::shared_ptr<module> module, bool clear_current_selection = true);
+    void select_module(std::vector<std::shared_ptr<module>> modules, bool clear_current_selection = true);
+
+    void select(std::shared_ptr<gate> gate, bool clear_current_selection = true);
+    void select(std::shared_ptr<net> net, bool clear_current_selection = true);
+    void select(std::shared_ptr<module> module, bool clear_current_selection = true);
+    void select(std::vector<std::shared_ptr<gate>> gates, bool clear_current_selection = true);
+    void select(std::vector<std::shared_ptr<net>> nets, bool clear_current_selection = true);
+    void select(std::vector<std::shared_ptr<module>> modules, bool clear_current_selection = true);
+    void select(std::vector<u32> gate_ids, std::vector<u32> net_ids, std::vector<u32> module_ids, bool clear_current_selection = true);
+    void select(std::vector<std::shared_ptr<gate>> gates, std::vector<std::shared_ptr<net>> nets, std::vector<std::shared_ptr<module>> modules, bool clear_current_selection = true);
+
+    void clear_gate(u32 gate_id);
+    void clear_gate(std::vector<u32> gate_ids);
+    void clear_gate(std::shared_ptr<gate> gate);
+    void clear_gate(std::vector<std::shared_ptr<gate>> gates);
+
+    void clear_net(u32 net_id);
+    void clear_net(std::vector<u32> net_ids);
+    void clear_net(std::shared_ptr<net> net);
+    void clear_net(std::vector<std::shared_ptr<net>> nets);
+
+    void clear_module(u32 module_id);  
+    void clear_module(std::vector<u32> module_ids );
+    void clear_module(std::shared_ptr<module> modul);
+    void clear_module(std::vector<std::shared_ptr<module>> modules);
+
+    void clear();
+    void clear(std::shared_ptr<gate> gate);
+    void clear(std::shared_ptr<net> net);
+    void clear(std::shared_ptr<module> module);
+    void clear(std::vector<std::shared_ptr<gate>> gates);
+    void clear(std::vector<std::shared_ptr<net>> nets);
+    void clear(std::vector<std::shared_ptr<module>> modules);
+    void clear(std::vector<u32> gate_ids, std::vector<u32> net_ids, std::vector<u32> module_ids );
+    void clear(std::vector<std::shared_ptr<gate>> gates, std::vector<std::shared_ptr<net>> nets, std::vector<std::shared_ptr<module>> modules);
```

Comment:
"Clear" is misleading, as it is normally used to remove all items from a collection.
"gui_api->clear" thus sounds like emptying the api :D
Use clear_selection or even better deselect